### ISSUE TITLE
Bug: Prevent creating draft projects with the same ID as a real project

### ISSUE THREAD

**Issue by myieye** [sillsdev/languageforge-lexbox#794]
We've only seen this happen due to [a bug that we fixed](https://github.com/sillsdev/languageforge-lexbox/issues/751), but theoretically a draft project could be created using the same ID as a real project. If that happens, when you try to delete the draft project you actually delete the real one.

At a minimum I'd at a check and throw an exception on the server to be sure it never happens.
Nice-to-have: On the Create project page, if the url passes in an ID for a project that already exists, we replace the form/page with: "This project has already been created: {#if admin}<project-code as link{/if}.

**Comment by hahn-kev** [comment]
I think we should also fix the draft delete so it only deletes a draft and never a real project.

**Comment by rmunn** [comment]
Yeah. When I implemented the draft delete I piggybacked off of the delete-project mutation because it was (supposed to be) impossible for a draft project to have the same GUID as a real one. Now that that's been proven to be possible after all, that design is flawed, and there needs to be a `deleteDraftProject` mutation.

**Comment by myieye** [comment]
Now that we use a seperate mutation for deleting draft projects, we don't have any real concerns.
And while there's a little bit of room for improvement here, I see no good reason to schedule improvements.

The worst that can happen is a slightly confusing situation.
Here's one way the "problem" could occur:
- A user requests a project
- The admin creates the project, but does not make the requesting user a **manager** (i.e. the user does not get the canCreateProjects permission) (previously, we had a bug that resulted in the user not getting the permission even if the admin made them a manager)
- The admin forwards the request email with the "Create project" link to the user
- The user clicks the "Create project" link taking them to the Create/Request project page
- The user requests the project again with the same ID

**Comment by myieye** [comment]
It's actually a bit easier for this to happen than I thought. Even if the admin makes the user a manager of the project, the user still doesn't have the canCreateProjects permission **until they re-login**. I've suggested to our admins that they don't forward the project request email to the user who requested the project, but I can imagine that it is nice to include that context if they need to discuss something with the user. I think we need a really simple fix for this.

We need to prevent creating a draft project with the same ID as an existing project (even if the existing project is already deleted), because it'll be impossible to turn that into a real project. And it would be great if the user is not left confused.
